File: go/grains/grains.go
package grains

import "errors"

// Square returns the number of grains on any given chessboard square
func Square(square int) (uint64, error) {
	if square <= 0 || square > 64 {
		return 0, errors.New("square must be between 1 and 64 (inclusive)")
	}
	return 1 << uint64(square-1), nil
}

// Total returns the number of grains on all squares
func Total() uint64 {
	return (1 << uint64(64)) - 1
}

File: typescript/grains/grains.ts
class Grains {
  static square(n: number): number {
    if (n < 1 || n > 64) {
      throw new Error("Invalid square.")
    }
    return 2 ** (n - 1)
  }

  static total(): number {
    return (2 ** 64) - 1
  }
}

export default Grains;